drivers/soundwire/intel_auxdevice.c

```diff
@@ -477,9 +477,12 @@ static void intel_link_remove(struct auxiliary_device *auxdev)
 	if (!bus->prop.hw_disabled) {
 		sdw_intel_debugfs_exit(sdw);
 		cancel_delayed_work_sync(&cdns->attach_dwork);
-		sdw_cdns_enable_interrupt(cdns, false);
```

Comment:
Don't we need at least disable the Peripheral interrupts? What if a jack event happens during the removal process?
IMHO, we should disable the interrupts except CDNS_MCP_INT_RX_WL here and disable the CDNS_MCP_INT_RX_WL interrupt after `sdw_bus_master_delete(bus);`

Reply:
Fair enough, thanks.